colossalai/inference/modeling/models/nopadding_llama.py

```diff
@@ -74,6 +83,9 @@ def llama_model_forward(
     sequence_lengths = batch.get_sequence_lengths()
     batch_size = batch.current_batch_size
     kv_seq_len = sequence_lengths.max().item()
+    use_cuda = True
+    if batch_size >= 32 and kv_seq_len > 512:
+        use_cuda = False
```

Comment:
Write some comments to explain the reason behind this if statement.

Reply:
Done